core/src/main/java/com/oath/oak/OakMap.java

```diff
@@ -52,12 +52,12 @@
 
     // internal constructor, to create OakMap use OakMapBuilder
     OakMap(K minKey, OakSerializer<K> keySerializer, OakSerializer<V> valueSerializer, OakComparator<K> oakComparator,
-           int chunkMaxItems, MemoryManager mm) {
+           int chunkMaxItems, MemoryManager mm, ValueUtils operator) {
```

Comment:
It is not a singleton if it does not have to be. Currently it is a singleton since it has no state, but if in the future some implementation would add a state, then it would help.
Moreover, it becomes more testable this way since mocks can be used if necessary.